apm/src/app/user/user.module.ts

```diff
@@ -0,0 +1,27 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { LoginComponent } from './login/login.component';
+import { UserComponent } from './user.component';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { RouterModule } from '@angular/router';
+import { UserService } from './user.service';
+
+@NgModule({
+  declarations: [
+    LoginComponent,
+    UserComponent,
+  ],
+  imports: [
+    CommonModule,
+    FormsModule,
+    ReactiveFormsModule,
+    RouterModule.forChild([
+      { path: 'login', component: LoginComponent},
+      { path: 'edit', component: UserComponent}
+    ])
+  ],
+  providers: [
+    UserService
```

Comment:
Just a note here again, for this service it does not matter if you had repeated the service registeration here and on the appModule. That is because the module is not lazy loaded and Angular will provide the same instance. But it would be different in case a module was lazy loaded. Just something to have in mind when you learn about lazy loading modules